pipeline/runtime/common.go

```diff
@@ -189,3 +191,19 @@ func GetReplacer(
 	wc := livelog.New(client, logKey, name, []logstream.Nudge{}, false)
 	return logstream.NewReplacer(wc, secrets)
 }
+
+func waitForFileWithTimeout(timeout time.Duration, filename string, fs filesystem.FileSystem) error {
```

Comment:
We dont need file for syncing in VM. We can use mutexes since there is one lite-engine context per stage. Check this usage and do something similar: https://github.com/harness/lite-engine/blob/ca535efeeb31348781da9d77dcc94130fe8b45c5/ti/config/config.go#L21

Reply:
fixed